### PR TITLE
Fix stream factory test

### PR DESCRIPTION
**Pull request by shadowhand**

### CODE REVIEW

**Comment by shadowhand** [comment]
I think this is the cause of hhvm failures, eg: https://travis-ci.org/http-interop/http-factory-diactoros/jobs/177855322

File: src/TestCase.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Http\FactoryTest;
+
+class TestCase extends \PHPUnit_Framework_TestCase
+{
+    protected function createTemporaryFile()
+    {
+        return tempnam(sys_get_temp_dir(), uniqid());
```
Comment (shadowhand) [inline src/TestCase.php:9]: For whatever reason, `tmpfile` fails on HHVM.

**Comment by ajgarlag** [comment]
The cause of failure is that zend-diactoros interprets that resources created with tmpfile are not readable because HHVM does not return a valid mode from stream metadata: https://3v4l.org/ThIDj

I think that this PR should fix the tests

**Comment by shadowhand** [comment]
See also facebook/hhvm#6670

**Comment by ajgarlag** [comment]
I've tested this PR in https://github.com/http-interop/http-factory-diactoros/pull/14 and it works successfully. You can see HHVM test result in https://travis-ci.org/http-interop/http-factory-diactoros/jobs/178218059

**Comment by shadowhand** [comment]
@http-interop/http-factory-contributors @Nyholm can someone approve this please?

**Pull request merged**